src/views/RouteTutkinnonosat.vue

```diff
@@ -130,6 +176,42 @@ export default class RouteTutkinnonosat extends Vue {
     await this.toteutussuunnitelmaStore.initNavigation();
   }
 
+  selectAllRows() {
+    this.valitseKaikki = !this.valitseKaikki;
+    if (this.valitseKaikki) {
+      this.selectedTutkinnonosat = [
+        ...this.selectedTutkinnonosat,
+        ...(this.tutkinnonosat || []) as SisaltoviiteLaajaDto[],
+      ];
+    }
+    else {
+      this.selectedTutkinnonosat = _.filter(this.selectedTutkinnonosat, sel => !_.includes(_.map(this.tutkinnonosat, 'tutkinnonosaViite.id'), _.get(sel, 'tutkinnonosaViite.id')));
```

Comment:
Jos valitsekaikki on false, eikö tässä pitäisi vain tyhjentää this.sekectedTutkinnonosat?

Reply:
Jos valitsekaikki on false niin silloin voi olla yksittäisiä kuitenkin valittuna.